ui/pages/ApplicationRemove.tsx

```diff
@@ -109,17 +109,15 @@ function ApplicationRemove({ className, name }: Props) {
   }, [name]);
 
   React.useEffect(() => {
-    if (repoRemoving || error) {
-      return;
-    }
-
+    if (!repoRemoveRes) return;
     const poll = poller(() => {
       applicationsClient
         .GetApplication({ name, namespace: "wego-system" })
-        .catch(() => {
-          // Once we get a 404, the app is gone for good
+        .catch((e) => {
           clearInterval(poll);
-          setRemovedFromCluster(true);
+          // Once we get a 404, the app is gone for good
+          if (e.code === GrpcErrorCodes.NotFound)
```

Comment:
Are you sure the backend will return this status code when it doesn't find the app? Might need to change it here:

https://github.com/weaveworks/weave-gitops/blob/1517a32c7806d3d62fc54e29c28d7e8e6cdb7a8a/pkg/server/server.go#L163

Something like:

```golang
		if apierrors.IsNotFound(err) {
                    return nil, grpcStatus.Errorf(codes.NotFound, "not found: %s", err.Error())
		}

```

Reply:
Was able to successfully remove app from cluster using this code - the app recognizes this error code and then shows the removed from cluster notification!